src/types/Species.ts

```diff
@@ -46,58 +46,27 @@ export const mockSpeciesNewFieldsData: Partial<Species> = {
 
 export type WoodDensityLevel = 'Species' | 'Genus' | 'Family';
 
-export type PlantMaterialSourcingMethod =
-  | 'Seed collection & germination'
-  | 'Seed purchase & germination'
-  | 'Mangrove propagules'
-  | 'Vegetative propagation'
-  | 'Wildling harvest'
-  | 'Seedling purchase'
-  | 'Other';
+export type PlantMaterialSourcingMethod = ArrayDeref<
+  NonUndefined<components['schemas']['GetSpeciesResponsePayload']['species']['plantMaterialSourcingMethods']>
+>;
 
-export type SuccessionalGroup = 'Pioneer' | 'Early secondary' | 'Late secondary' | 'Mature';
+export type SuccessionalGroup = ArrayDeref<
+  NonUndefined<components['schemas']['GetSpeciesResponsePayload']['species']['successionalGroups']>
+>;
 
 export type NativeStatus = 'Native' | 'Non-Native';
 
-export type EcosystemType =
-  | 'Boreal forests/Taiga'
-  | 'Deserts and xeric shrublands'
-  | 'Flooded grasslands and savannas'
-  | 'Mangroves'
-  | 'Mediterranean forests, woodlands and scrubs'
-  | 'Montane grasslands and shrublands'
-  | 'Temperate broad leaf and mixed forests'
-  | 'Temperate coniferous forest'
-  | 'Temperate grasslands, savannas and shrublands'
-  | 'Tropical and subtropical coniferous forests'
-  | 'Tropical and subtropical dry broad leaf forests'
-  | 'Tropical and subtropical grasslands, savannas and shrublands'
-  | 'Tropical and subtropical moist broad leaf forests'
-  | 'Tundra';
+export type EcosystemType = ArrayDeref<
+  NonUndefined<components['schemas']['GetSpeciesResponsePayload']['species']['ecosystemTypes']>
+>;
 
-export type GrowthForm =
-  | 'Tree'
-  | 'Shrub'
-  | 'Forb'
-  | 'Graminoid'
-  | 'Fern'
-  | 'Fungus'
-  | 'Lichen'
-  | 'Moss'
-  | 'Vine'
-  | 'Liana'
-  | 'Shrub/Tree'
-  | 'Subshrub'
-  | 'Multiple Forms';
+export type GrowthForm = ArrayDeref<
+  NonUndefined<components['schemas']['GetSpeciesResponsePayload']['species']['growthForms']>
+>;
 
-export type SeedStorageBehavior =
-  | 'Intermediate'
-  | 'Likely Intermediate'
-  | 'Likely Orthodox'
-  | 'Likely Recalcitrant'
-  | 'Orthodox'
-  | 'Recalcitrant'
-  | 'Unknown';
+export type SeedStorageBehavior = NonUndefined<
```

Comment:
I think you can use Required<> here, like in https://github.com/terraware/terraware-web/blob/main/src/types/Accession.ts#L9

Reply:
That didn't seem to work unfortunately:

![SCR-20240426-nmye.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/wJnPX9wLZAkiGXEBFJxL/359bec5b-aa59-4419-a8d7-66241e108285.png)